### PR TITLE
DAOS-11028 control: Allow daos_server network scan run without config…

### PR DESCRIPTION
**Pull request by wiliamhuang**
… file

"daos_server network scan" should be able to run even without a configuration file.

Signed-off-by: Michael MacDonald <mjmac.macdonald@intel.com>
Signed-off-by: Lei Huang <lei.huang@intel.com>

### CODE REVIEW

File: src/control/cmd/daos_server/main.go [hunk 1]
```diff
@@ -152,7 +152,9 @@ func parseOpts(args []string, opts *mainOpts, log *logging.LeveledLogger) error
 			if err := cfgCmd.loadConfig(opts.ConfigPath); err != nil {
 				return errors.Wrapf(err, "failed to load config from %s", cfgCmd.configPath())
 			}
-			log.Infof("DAOS Server config loaded from %s", cfgCmd.configPath())
+			if _, err := os.Stat(opts.ConfigPath); err == nil {
```
Comment (knard-intel) [inline src/control/cmd/daos_server/main.go:155]: I was expecting to have this information message in your new function `func (c *optCfgCmd) loadConfig(cfgPath string) error`.

**Review by knard-intel — COMMENTED** [review]
From my understanding we will not have any more clear error message for subcommand needing a configuration file ?

Some unit tests could also be helpful validating that subcommand needing configuration file and subcommand doesn't needing it are well managed.

File: src/control/cmd/daos_server/main.go (hunk 1 above)
Comment (wiliamhuang) [reply src/control/cmd/daos_server/main.go:155]: Then we need to print the information both in "func (c *cfgCmd) loadConfig(cfgPath string)" and "func (c *optCfgCmd) loadConfig(cfgPath string)". Do you prefer this way?

**Comment by wiliamhuang** [comment]
> From my understanding we will not have any more clear error message for subcommand needing a configuration file ?
> It could be helpful to have some unit tests testing the management of configuration file missing according to the subcommand used.

This PR targets "daos_server network " only. I believe other subcommands which require a configuration file will fail in case a configuration file does not exist. "failed to load config from xxxx" will be printed out.

File: src/control/cmd/daos_server/main.go (hunk 1 above)
Comment (knard-intel) [reply src/control/cmd/daos_server/main.go:155]: Do not bother too much with that, I am not a GO expert and thus I am not sure if that is the best coding practice.

**Review by knard-intel — APPROVED** [review]
LGFM with latest explanation regarding the management of configuration file not present with subcommand different from scan.

**Review by mjmac — APPROVED** [review]

**Comment by wiliamhuang** [comment]
@mjmac @knard-intel I removed one blank line at the end of src/control/cmd/daos_server/config.go to fix a CI issue. I also merged with latest master. Now all CI tests passed. Could you please approve this PR? Thank you!

**Review by knard-intel — APPROVED** [review]

**Review by mjmac — APPROVED** [review]

**Pull request merged**